components/clarinet-files/src/lib.rs

```diff
@@ -18,11 +18,31 @@ pub use project_manifest::{
     ContractConfig, ProjectManifest, ProjectManifestFile, RequirementConfig,
 };
 use serde::ser::{Serialize, SerializeMap, Serializer};
+use std::future::Future;
+use std::pin::Pin;
 use std::{borrow::BorrowMut, path::PathBuf, str::FromStr};
 use url::Url;
 
 pub const DEFAULT_DEVNET_BALANCE: u64 = 100_000_000_000_000;
 
+pub type PerformFileAccess = Pin<Box<dyn Future<Output = Result<(FileLocation, String), String>>>>;
+pub type PerformFileWrite = Pin<Box<dyn Future<Output = Result<(), String>>>>;
+
+pub trait FileAccessor {
```

Comment:
@lgalabru 
Do we still need a trait for `FileAccessor`? 
The original idea was to have a Native and a Wasm FileAccessor, but since we only have a wasm one, I'm not sure we still need the trait. wdyt?

Reply:
Considering the simplicity it brings, I'd rather remove it now, and re-implement the trait in the future if/when needed

Reply:
Attempt to remove the trait here: https://github.com/hirosystems/clarinet/commit/329ef9cc0fe878f196fa00dec2032f88dd1910b8

It's brings simplification and is easy to do but lead to unexpected issue: the WASM FS Accessor can't be used in place of the FileAccessor trait in non-wasm context